src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -725,8 +718,7 @@ internal void FinishOperationSuccess(SocketError socketError, int bytesTransferr
                     // Mark socket connected.
                     if (socketError == SocketError.Success)
                     {
-                        if (s_loggingEnabled)
-                            SocketsEventSource.Connected(_currentSocket, _currentSocket.LocalEndPoint, _currentSocket.RemoteEndPoint);
+                        NetEventSource.Connected(_currentSocket, _currentSocket.LocalEndPoint, _currentSocket.RemoteEndPoint);
```

Comment:
One additional thought:

There's little potential confusion in trivial cases like `Enter(this)` or `Error(this, e)`.  So perhaps a middle path would be to keep the convenience methods for these cases and not require wrapping them in if(enabled) tests, while requiring anything more involved than this to do an explicit if(enabled) test.

For clarity, though, I think I'd name these convenience methods something other than just "Enter" and "Exit".  Like"CheckEnabledAndEnter".  That's a terrible name of course, but you get the idea.